lua/nvim-tree/actions/git/init.lua

```diff
@@ -0,0 +1,343 @@
+      events._dispatch_will_rename_node(path_src, path_dst)
+      M.prompt_for_rename(node, { absolute = true })
+      -- Skip git mv
+      goto loop
+    end
+
+    events._dispatch_will_rename_node(path_src, path_dst)
+    M.git_mv(path_src, path_dst, function(err, data)
+      if err then
+        log.line("action_git", "paste has failed %", err)
+        notify.error(("git mv failed: %s"):format(data))
+      elseif data then
+        -- git may return error or fatal via stderr
+        if data:match "error:.*" or data:match "fatal:.*" then
+          log.line("action_git", "paste has failed %", data)
+          notify.error(("git mv failed: %s"):format(data))
+        else
+          clip_pasted = clip_pasted + 1
+          notify.info(("git mv - success: '%s' -> '%s'"):format(path_src, path_dst))
+
+          utils.rename_loaded_buffers(path_src, path_dst)
+          events._dispatch_node_renamed(path_src, path_dst)
+        end
+      end
+    end) -- git mv
+
+    ---@diagnostic disable-next-line: unused-label
+    ::loop::
+  end -- loop
+
+  -- paste everything in fs copy-paste clipboard
+  actions_fs_cpp.paste(node)
+
+  if clip_pasted > 0 then
+    notify.info(("Clipboard: pasted %s paths"):format(clip_pasted))
+    M.pasted = clip_pasted
+  end
+  M.clipboard.clear()
+
+  if not M.config.filesystem_watchers.enable then
+    reloaders.reload_explorer()
+  end
+end -- M.paste
+
+--- Options for prompt_for_rename
+--- At least one field must be specified
+--- @class RenamePromptOpts
+--- @field basename boolean|nil
+--- @field absolute boolean|nil
+--- @field filename boolean|nil
+--- @field filename_omit boolean|nil
+
+--- Prompt user to specify new path by using UI prompt and then use git for rename
+--- if git fails, fallback to actions.actions.fs.rename-file routine
+--- @param node table nvim-tree node instance
+--- @param opts RenamePromptOpts options to tweak prompt
+function M.prompt_for_rename(node, opts)
+  if not (opts and (opts.filename or opts.basename or opts.absolute or opts.filename_omit)) then
+    error "git actions: opts are missing or lack required field"
+  end
+
+  if type(node) ~= "table" then
+    node = lib.get_node_at_cursor()
+  end
+
+  node = lib.get_last_group_node(node)
+  if node.name == ".." then
+    return
+  end
+
+  local fnamemodify = vim.fn.fnamemodify
+  local path_sep = package.config:sub(1, 1)
+  local path_src_basename_len = node.name:len()
+  local path_src_ext = fnamemodify(node.name, ":e")
+  path_src_ext = #path_src_ext == 0 and "" or "." .. path_src_ext
+  local path_src_dirname = node.absolute_path:sub(0, path_src_basename_len * -1 - 1)
+
+  -- Set prompt placeholder
+  local prompt_default = ""
+  prompt_default = opts.absolute and node.absolute_path or prompt_default
+  prompt_default = opts.filename and fnamemodify(node.absolute_path, ":t") or prompt_default
+  prompt_default = opts.basename and fnamemodify(node.absolute_path, ":t:r") or prompt_default
```

Comment:
This method repeats a lot of `rename-file.lua`.

Can we please share?

Reply:
It can be shared by a separate `refactor(actions): rename-file.lua` PR. Are you going to accept a separate PR with `rename-file.lua` refactoring? 

Currently `rename-file.lua` isn't exporting internal utils so it's not reusable. The chunk of code above si similar to (IMO very inefficient) following string comparison:
https://github.com/nvim-tree/nvim-tree.lua/blob/46e1f776f0a714099b9160604fbb62fba2540376/lua/nvim-tree/actions/fs/rename-file.lua#L64-L79

Reply:
> It can be shared by a separate `refactor(actions): rename-file.lua` PR. Are you going to accept a separate PR with `rename-file.lua` refactoring?

Absolutely! That's what I meant by "prefactor". Sorry, I shoud have been clearer.

Something like "chore(#2542): refactor rename file"

Reply:
Or `refactor(…): …`. :)

Reply:
Heh... didn't realise that was a prefix, thank you.

Reply:
Please merge:
https://github.com/nvim-tree/nvim-tree.lua/pull/2550

This PR is going to use common code from that PR.